solr/core/src/test/org/apache/solr/search/TestExtendedDismaxParser.java

```diff
@@ -1776,6 +1764,36 @@ private static String getParsedQuery(SolrQueryRequest request) throws Exception
     return (String) BaseTestHarness.evaluateXPath(resp, "//str[@name='parsedquery']/text()", XPathConstants.STRING);
   }
   
+  public void testSplitOnWhitespace_shouldRespectMinimumShouldMatch(){
```

Comment:
I think I've done it, formatting is one of the things I hate the most, my opinion is that it shouldn't be the responsibility of single peers but it should be server side to avoid inconsistencies(which I often find also in Lucene/Solr and other Open Source code).
Never really explored a solution for that either, so I hope, my changes are ok now :)

Reply:
https://issues.apache.org/jira/browse/SOLR-14920 -- I anticipate this will happen for 9.0 soon.